__init__.py

```diff
@@ -55,89 +74,103 @@ def initialize(self):
-        self.gui["prompt2"] = "Follow the \nprompt on your \nmobile device or \ncomputer"
-        self.gui.show_page("prompt2.qml")
-        wait_while_speaking()
-        self.speak_dialog("2b_sign.in.to.ap_speech")
-        self.gui["prompt2"] = "If you don't get \na prompt, go to \nstart.mycroft.ai"
-        self.gui["phone_image2"] = "3_phone_choose-wifi.png"
-
-    def prompt_to_select_network(self, _=None):
+        self._show_page("follow_prompt")
+        self.speak_dialog("choose-wifi-network", wait=True)
+        self.speak_dialog("no-prompt", wait=True)
+
+    def _display_select_wifi_network(self):
         """Prompt user to select network and login."""
-        self.gui["phone_image3"] = "3_phone_choose-wifi.png"
-        self.gui["prompt3"] = "Choose the \nWifi network to \nconnect your \nMycroft device"
-        self.gui.show_page("prompt3.qml")
+        self._show_page("network_select")
 
-    def check_connection(self):
-        is_connected = connected()
-        if is_connected:
+    def _check_for_internet_connection(self):
+        """Determine if the device connected successfully."""
+        self.connected_to_internet = connected()
+        if self.connected_to_internet:
             stop_speaking()
-            self.report_setup_complete()
-        return is_connected
 
-    def report_setup_complete(self, _=None):
+    def _report_setup_complete(self):
         """Report when wifi setup is complete, network is connected."""
-        # self.speak_dialog("4_internet.connected_speech")
-        self.gui["bgColor"] = green
-        self.gui["icon"] = "check-circle.svg"
-        self.gui["label"] = self.translate("4_internet.connected_screen")
-        self.gui.remove_page("prompt1.qml")
-        self.gui.remove_page("prompt2.qml")
-        self.gui.remove_page("prompt3.qml")
-        self.gui.show_page("status.qml")
-        # wait_while_speaking()
+        self.gui["label"] = self.translate("connected")
+        self._show_page("wifi_success")
         sleep(5)
         self.gui.release()
         self.bus.emit(Message('mycroft.ready'))
 
-    def report_error(self, _=None):
-        """Report if an error occured during wifi setup."""
-        self.gui.remove_page("prompt.qml")
-        self.gui["bgColor"] = red
-        self.gui["icon"] = "times-circle.svg"
-        self.gui["label"] = "Incorrect password"
-        self.gui.show_page("status.qml")
+    def _show_page(self, page_name_prefix: str):
+        """Shows the appropriate screen for the device's platform.
+
+        Args:
+            page_name_prefix: part of the page name not platform-specific
+        """
+        if self.gui.connected:
+            if self.platform == MARK_II:
+                page_name_suffix = "_mark_ii"
+            else:
+                page_name_suffix = "_scalable"
+            page_name = page_name_prefix + page_name_suffix + ".qml"
+            if self.page_showing is not None:
+                self.gui.remove_page(self.page_showing)
```

Comment:
I'd like to keep the pagination in until we get the pages triggered from awconnect events. This means people can swipe back to few previous setup steps.

Reply:
Pagination has crashed my GUI on multiple occasions.  I don't think we want the GUI to crash when people are setting up their device.

Reply:
We really need to dig into that and fix it properly. I've had it when there were lots of assets all loading and animating but not for super simple things like a few static pages.

I'm not going to block the PR for it though.